chainerrl/misc/prioritized.py

```diff
@@ -0,0 +1,197 @@
+import random
+
+
+class PrioritizedBuffer (object):
+    def __init__(self, capacity=None):
+        self.capacity = capacity
+        self.data = []
+        self.priority_tree = SumTree()
+        self.data_inf = []
+        self.flag_wait_priority = False
+
+    def __len__(self):
+        return len(self.data) + len(self.data_inf)
+
+    def append(self, value):
+        # new values are the most prioritized
+        self.data_inf.append(value)
+        if len(self) > self.capacity:
+            self.pop()
+
+    def _pop_random_data_inf(self):
+        assert self.data_inf
+        n = len(self.data_inf)
+        i = random.randrange(n)
+        ret = self.data_inf[i]
+        self.data_inf[i] = self.data_inf[n-1]
+        self.data_inf.pop()
+        return ret
+
+    def pop(self):
+        """Remove an element uniformly.
+
+        Not prioritized.
+        """
+        assert len(self) > 0
+        assert not self.flag_wait_priority
+        n = len(self.data)
+        if n == 0:
+            return self._pop_random_data_inf()
+        i = random.randrange(0, n)
+        # remove i-th
+        self.priority_tree[i] = self.priority_tree[n-1]
+        del self.priority_tree[n-1]
+        ret = self.data[i]
+        self.data[i] = self.data.pop()
```

Comment:
> `self.data[i] = self.data.pop()`

This would raise an out-of-range error if i == n - 1 because self.data has only n-1 elements after pop.